src/redux/session/reducer/signUpReducer.js

```diff
@@ -0,0 +1,116 @@
+import { getRefreshToken, setRefreshToken, deleteRefreshToken } from '../helpers';
+import {
+  SIGN_UP_SUCCESS, SIGN_UP_FAILURE, REFRESH_ACCESS_TOKEN_SUCCESS, REFRESH_ACCESS_TOKEN_FAILURE,
+  LOGIN_SUCCESS, LOGIN_FAILURE, LOGOUT_SUCCESS, LOGOUT_FAILURE, CLEAR_ERROR,
+} from '../action/sessionActions';
+
+const initialState = {
+  currentUser: {
+    id: null,
+    email: null,
+    role: null,
+    createdAt: null,
+  },
+  error: false,
+  loading: true,
+  errorMsgs: '',
+  accessToken: null,
+  refreshToken: getRefreshToken(),
+  expiresIn: null,
+  tokenType: null,
+};
+
```

Comment:
I would suggest you refactor this whole code since most of the functions here are redundant.